File: MyWebApp/AcessaDados/Entidades/Produto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace MyWebApp.AcessaDados.Entidades
{
    public class Produto
    {
        public int Id { get; set; }
        public string descricao { get; set; }
        public string nome { get; set; }
        public decimal preco { get; set; }
    }
}

File: MyWebApp/wwwroot/ts/produto.ts
class Produto {
    public id: number = 0;
    public nome: string

    public obterNome(): string {
        return this.nome;
    }

    public obterID(): number {
        return this.id;
    }

    set descricao(val) {

    }

    get descricao() {
        return this.nome;
    }
}